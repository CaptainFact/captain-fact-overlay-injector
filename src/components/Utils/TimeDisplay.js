import React from 'react'
import padLeft from 'voca/pad_left'
import classNames from 'classnames'

import { link } from './TimeDisplay.css'


function formatSeconds(totalSeconds) {
  if (!totalSeconds)
    return '0:00:00'

  const sign = totalSeconds < 0 ? '-' : ''
  const absSeconds = Math.abs(totalSeconds)
  const hours = Math.trunc(absSeconds / 3600)
  const minutes = Math.trunc((absSeconds / 60) % 60)
  const seconds = padLeft(absSeconds % 60, 2, '0')

  return `${sign}${hours}:${padLeft(minutes, 2, '0')}:${seconds}`
}

const TimeDisplay = ({ time, handleClick, className,  textBefore = '' }) => {
  const formattedTime = formatSeconds(time)
  const content = handleClick ? (
    <a
      onClick={() => handleClick(time)}
      className={classNames(link, className)}
    >
      {formattedTime}
    </a>
  ) : formattedTime

  return <span>{textBefore}{content}</span>
}

export default TimeDisplay
