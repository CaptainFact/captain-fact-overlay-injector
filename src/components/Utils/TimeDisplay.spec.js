import React from 'react'
import TimeDisplay from './TimeDisplay'


test('render time', () => {
  snapshot(<TimeDisplay time={0}/>)
  snapshot(<TimeDisplay time={42}/>)
  snapshot(<TimeDisplay time={60}/>)
  snapshot(<TimeDisplay time={3600}/>)
  snapshot(<TimeDisplay time={7342}/>)
})

test('add a - sign if negative', () => {
  snapshot(<TimeDisplay time={-42}/>)
})

test('prefix with given text', () => {
  snapshot(<TimeDisplay time={42} textBefore="Dont't feed greemlins after "/>)
})

test('time becomes clickable if a function is passed', () => {
  snapshot(<TimeDisplay time={1334} handleClick={() => 'Awesome'}/>)
})

test('handleClick gets called with time', () => {
  const time = 4242
  const clickFunc = jest.fn()
  const mounted = shallow(<TimeDisplay time={time} handleClick={clickFunc}/>)
  const link = mounted.find('a').first()
  link.simulate('click')
  expect(clickFunc).toBeCalledWith(time)
})