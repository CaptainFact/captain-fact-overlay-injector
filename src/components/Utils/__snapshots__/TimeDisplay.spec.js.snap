// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add a - sign if negative 1`] = `
<span>
  -0:00:42
</span>
`;

exports[`prefix with given text 1`] = `
<span>
  Dont't feed greemlins after 
  0:00:42
</span>
`;

exports[`render time 1`] = `
<span>
  0:00:00
</span>
`;

exports[`render time 2`] = `
<span>
  0:00:42
</span>
`;

exports[`render time 3`] = `
<span>
  0:01:00
</span>
`;

exports[`render time 4`] = `
<span>
  1:00:00
</span>
`;

exports[`render time 5`] = `
<span>
  2:02:22
</span>
`;

exports[`time becomes clickable if a function is passed 1`] = `
<span>
  <a
    className="link"
    onClick={[Function]}
  >
    0:22:14
  </a>
</span>
`;
